Compile/src/Producers/LeptonSFProducer.cc

```diff
@@ -468,3 +470,95 @@ void LeptonTriggerSFProducer::Produce(ZJetEvent const& event,
         }
     }
 }
+
+/////////////////////
+// LeptonRecoSF //
+/////////////////////
+
+std::string LeptonRecoSFProducer::GetProducerId() const { return "LeptonRecoSFProducer"; }
+
+void LeptonRecoSFProducer::Init(ZJetSettings const& settings)
+{
+    m_sffile = settings.GetLeptonRecoSFRootfile();
+    if (settings.GetLeptonSFVariation() == true) {
+        LOG(WARNING) << "LeptonRecoSFProducer: varying scale factor UP and DOWN one sigma";
+    }
+
+    if(settings.GetChannel() == "mm"){
```

Comment:
Why is this only for the Muon channel? Aren't there also SFs for Electrons?

Reply:
Maybe we should then think of a change in the naming. When all the Producers only produce Muons, it makes no sense to designate them as Lepton-Producers.

Reply:
Theoretically it should work fine for electrons and gammas as well. For e and gamma there are only ID and Reco efficiencies though: https://twiki.cern.ch/twiki/bin/view/CMS/EgammaUL2016To2018#SFs_for_Electrons_UL_2018

Reply:
Okay. I would recommend then to also add these scale factors in the producers, where valid. For the others, add a warning which prints, that there is no recommendation for this SF and this will be skipped.

Reply:
I don't have the time to add them currently. I think doing so once they are needed is fine as well.

Reply:
Okay, I will open an issue for this to keep track.

Reply:
see #81